graphql/mutation_resolver.go

```diff
@@ -108,6 +108,37 @@ func (r *mutationResolver) RemoveAnnotationIssue(ctx context.Context, taskID str
 	}
 }
 
+// AddAnnotationMetadataLink is the resolver for the addAnnotationMetadataLink field.
+func (r *mutationResolver) AddAnnotationMetadataLink(ctx context.Context, taskID string, execution int, apiMetadataLink restModel.APIMetadataLink) (bool, error) {
```

Comment:
Assuming we are confident we won't change our minds then one mutation resolver that lets you edit the link makes sense to me. 

If that's the case we should update this on the DB side as well to accept a field instead of an array.

Reply:
If we still want to enable extending the number of links a user can add in the future I'm fine to do this while keeping the field on the annotation as a list struct.  I assume what makes the most sense is accepting a list of `MetadataLink` into this resolver.